data/data/bootstrap/baremetal/files/usr/local/bin/startironic.sh.template

```diff
@@ -152,11 +149,47 @@ $IPTABLES -t raw -A DHCP_IRONIC -j DROP
 {{end}}
 
 # Wait for images to be downloaded/ready
-podman wait -i 1000ms ipa-downloader
 podman wait -i 1000ms coreos-downloader
-while ! curl --fail http://localhost/images/rhcos-ootpa-latest.qcow2.md5sum ; do sleep 1; done
-while ! curl --fail --head http://localhost/images/ironic-python-agent.initramfs ; do sleep 1; done
-while ! curl --fail --head http://localhost/images/ironic-python-agent.kernel ; do sleep 1; done
+
+export KUBECONFIG=/opt/openshift/auth/kubeconfig-loopback
+
+mkdir -p /tmp/nmstate
+
+{{range .PlatformData.BareMetal.Hosts}}
+    until oc get -n openshift-machine-api baremetalhost {{.Name}}; do
+        echo Waiting for Host {{.Name}} to appear...
+        sleep 10
+    done
+    secret_name=$(oc get -n openshift-machine-api baremetalhost {{.Name}} -o jsonpath="{.spec.preprovisioningNetworkDataName}")
+    if [ -n "${secret_name}" ]; then
+        until oc get -n openshift-machine-api secret "${secret_name}"; do
+            echo Waiting for Secret "${secret_name}" to appear...
+            sleep 10
+        done
+        oc get -n openshift-machine-api secret ${secret_name} -o jsonpath="{.data.nmstate}" | base64 -d > /tmp/nmstate/{{.Name}}.yaml
+    else
+        touch /tmp/nmstate/{{.Name}}.yaml
+    fi
+{{end}}
+
```

Comment:
@imain I think that this whole block could be simplified, given that the `NetworkConfig` field is already available as a template parameter. Ie: 

```
{{range .PlatformData.BareMetal.Hosts}}
  echo "{{.NetworkConfig}}" > /tmp/nmstate/{{.Name}}.yaml
{{end}}
```

Reply:
That's actually what I had before.  Steve wanted to use the network secrets to get the data.  Apparently they could be modified by the user after configuration and before/during deployment.

Reply:
@hardys just to clarify, do you have more details about this point? In understood that the main purpose of the secret was for day-2 operation.
During the installation the user is expected to provide the configuration in the install-config.yaml, from where the `NetworkConfig` field is extracted